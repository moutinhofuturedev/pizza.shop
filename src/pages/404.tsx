import { useNavigate } from 'react-router-dom'

import { Button } from '@/components/ui/button'

export const NotFound = () => {
  const navigate = useNavigate()

  return (
    <div className="flex h-screen flex-col items-center justify-center">
      <div className="mx-auto max-w-screen-xl px-4 py-8 max-md:py-16 lg:px-6">
        <div className="mx-auto max-w-screen-sm text-center">
          <h1 className="mb-4 text-5xl font-extrabold tracking-tight text-muted-foreground max-lg:text-7xl">
            Whoops!404
          </h1>
          <p className="tracking-tigh mb-4 text-lg font-bold text-muted-foreground dark:text-white max-lg:text-4xl">
            Página não encontrada.
          </p>
          <Button onClick={() => navigate('/')}>Volte para o Dashboard</Button>
        </div>
      </div>
    </div>
  )
}
