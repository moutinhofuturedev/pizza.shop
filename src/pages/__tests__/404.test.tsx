import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { MemoryRouter } from 'react-router-dom'
import { NotFound } from '../404'

describe('<404 />', () => {
	it('should render correctly', () => {
		const { container } = render(<NotFound />, {
			wrapper: ({ children }) => (
				<MemoryRouter initialEntries={['/404']}>{children}</MemoryRouter>
			),
		})

		expect(container).toMatchSnapshot()
	})

	it('should render 404 message', () => {
		render(<NotFound />, {
			wrapper: ({ children }) => (
				<MemoryRouter initialEntries={['/404']}>{children}</MemoryRouter>
			),
		})

		const message1 = screen.getByText('Whoops!404')
		const message2 = screen.getByText('Página não encontrada.')

		expect(message1).toBeInTheDocument()
		expect(message2).toBeInTheDocument()
	})

	it('should render 404 message with button back to home', async () => {
		render(<NotFound />, {
			wrapper: ({ children }) => (
				<MemoryRouter initialEntries={['/404']}>{children}</MemoryRouter>
			),
		})

		const button = screen.getByRole('button', {
			name: /volte para o dashboard/i,
		})
		await userEvent.click(button)

		expect(window.location.pathname).toBe('/')
	})
})
