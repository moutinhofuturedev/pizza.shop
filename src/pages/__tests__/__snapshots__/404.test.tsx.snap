// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<404 /> > should render correctly 1`] = `
<div>
  <div
    class="flex h-screen flex-col items-center justify-center"
  >
    <div
      class="mx-auto max-w-screen-xl px-4 py-8 max-md:py-16 lg:px-6"
    >
      <div
        class="mx-auto max-w-screen-sm text-center"
      >
        <h1
          class="mb-4 text-5xl font-extrabold tracking-tight text-muted-foreground max-lg:text-7xl"
        >
          Whoops!404
        </h1>
        <p
          class="tracking-tigh mb-4 text-lg font-bold text-muted-foreground dark:text-white max-lg:text-4xl"
        >
          Página não encontrada.
        </p>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
        >
          Volte para o Dashboard
        </button>
      </div>
    </div>
  </div>
</div>
`;
